File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    words = {'zero': 0, 'one': 1, 'two': 2,
             'three': 3, 'four': 4, 'five': 5,
             'six': 6, 'seven': 7, 'eight': 8, 'nine': 9}

    idx = 0
    answer = 0

    while idx < len(s):
        if s[idx].isdigit():
            answer = answer * 10 + int(s[idx])
            idx += 1
            continue
        for i in range(3, 6):
            if s[idx: idx+i] in words:
                answer = answer * 10 + words[s[idx: idx+i]]
                idx += i
                break

    return answer

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.cpp
#include <string>
#include <cctype>
#include <vector>

using namespace std;

int solution(string s) {
    vector<string> nums = {"zero", "one", "two", "three", "four",
                           "five", "six", "seven", "eight", "nine"};

    int answer = 0;
    int idx = 0;

    while (idx < s.length()) {
        if (isdigit(s[idx])) {
            answer = answer * 10 + s[idx] - '0';
            idx++;
        } else {
            for (int i = 3; i < 6; i++) {
                string chk = s.substr(idx, i);
                bool flag = false;
                for (int j = 0; j < 10; j++) {
                    if (chk == nums[j]) {
                        flag = true;
                        answer = answer * 10 + j;
                        idx += i;
                        break;
                    }
                }
                if (flag) break;
            }
        }
    }
    
    return answer;
}